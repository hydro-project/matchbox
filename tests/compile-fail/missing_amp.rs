#[derive(PartialEq, Eq, Debug, Clone)]
enum LispValue {
    Nil,
    Cons(Box<LispValue>, Box<LispValue>),
    Symbol(String),
}
use LispValue::*;

fn main() {
    matchbox::matchbox! {
        match LispValue::Nil {
            Nil => {},
            Cons(_, _) => {},
            Symbol(mb!(** s)) => {},
            _ => panic!(),
        }
    }
}
