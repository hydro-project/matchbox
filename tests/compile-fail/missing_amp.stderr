error: expected one of: identifier, `::`, `<`, `_`, literal, `const`, `ref`, `mut`, `&`, parentheses, square brackets, `..`, `const`
  --> tests/compile-fail/missing_amp.rs:14:25
   |
14 |             Symbol(mb!(** s)) => {},
   |                         ^

warning: unreachable pattern
  --> tests/compile-fail/missing_amp.rs:15:13
   |
15 |             _ => panic!(),
   |             ^
   |
   = note: `#[warn(unreachable_patterns)]` on by default
